### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the main source doc for this Elixir
library which leverage on latest ExDoc features.

### CODE REVIEW

**Comment by benwilson512** [comment]
Great stuff, thank you!

File: .formatter.exs [hunk 1]
```diff
@@ -1,4 +1,5 @@
+# Used by "mix format"
 [
-  inputs: ["mix.exs", "{config,lib,test}/**/*.{ex,exs}"],
```
Comment (dolfinus) [inline .formatter.exs:2]: Well, this last comma should not be removed. Now `mix format` fails with an error:
```
** (SyntaxError) .formatter.exs:3:3: syntax error before: import_deps
    (elixir 1.11.3) lib/code.ex:340: Code.eval_string_with_error_handling/3
    (mix 1.11.3) lib/mix/tasks/format.ex:310: Mix.Tasks.Format.eval_file_with_keyword_list/1
    (mix 1.11.3) lib/mix/tasks/format.ex:172: Mix.Tasks.Format.eval_dot_formatter/1
    (mix 1.11.3) lib/mix/tasks/format.ex:141: Mix.Tasks.Format.run/1
    (mix 1.11.3) lib/mix/task.ex:394: Mix.Task.run_task/3
    (mix 1.11.3) lib/mix/cli.ex:84: Mix.CLI.run_task/2
```

**Comment by benwilson512** [comment]
Ah, we should add the format check to the CI. Will fix in master.

**Comment by dolfinus** [comment]
Well, it is already done :) https://github.com/absinthe-graphql/absinthe_relay/pull/181

**Pull request merged**